pyuvsim/source.py

```diff
@@ -145,48 +189,45 @@ def alt_az_calc(self, time, telescope_location):
             raise ValueError('telescope_location must be an astropy EarthLocation object. '
                              'value was: {al}'.format(al=telescope_location))
 
-        source_altaz = self.skycoord.transform_to(AltAz(obstime=time, location=telescope_location))
+        if self.time == time:  # Don't repeat calculations
+            return
 
-        alt_az = (source_altaz.alt.rad, source_altaz.az.rad)
-        self.time = time
-        self.alt_az = alt_az
-        return alt_az
+        skycoord_use = SkyCoord(self.ra, self.dec, frame='icrs')
+        source_altaz = skycoord_use.transform_to(AltAz(obstime=time, location=telescope_location))
 
-    def get_alt_az(self, time, telescope_location):
-        """ Reuse alt_az if already calculated """
-        if (self.alt_az is None) or (not time == self.time):
-            return self.alt_az_calc(time, telescope_location)
-        else:
-            return self.alt_az
+        time.location = telescope_location
+        lst = time.sidereal_time('apparent')
 
-    def pos_lmn(self, time, telescope_location):
-        """
-        calculate the direction cosines of this source at a time & location
+        cirs_source_coord = skycoord_use.transform_to('cirs')
+        tee_ra = simutils.cirs_to_tee_ra(cirs_source_coord.ra, time)
 
-        Args:
-            time: astropy Time object
-            telescope_location: astropy EarthLocation object
+        self.hour_angle = None
```

Comment:
Where is hour_angle used? I'd prefer to stick with alt/az

Reply:
`hour_angle` comes up in `coherency_calc` (lines 152-153)

Reply:
ah, ok. That's going away with the polarized branch, but we can leave it for now.